lang/csharp/src/apache/codegen/AvroGen.cs

```diff
@@ -193,5 +194,49 @@ static void Usage()
 
             return 0;
         }
+
+        /// <summary>
+        /// Replace namespace(s) in schema or protocol definition.
+        /// </summary>
+        /// <param name="input">input schema or protocol definition.</param>
+        /// <param name="namespaceMapping">namespace mappings object.</param>
+        public static string ReplaceMappedNamespacesInSchema(string input, IEnumerable<KeyValuePair<string, string>> namespaceMapping)
```

Comment:
Does this method need to be `public` ?

Reply:
Not really. I think my original intention was to add unit test to test specifically the replace code. However I decided not to, since there is "high" level tests, which indirectly test this function. I will go ahead and make it private.